File: solutions/go/raindrops/1/raindrops.go
package raindrops

import (
    "strconv"
    "strings"
    )
    

func Convert(number int) string {
    var sb strings.Builder
    var factor bool = false

    if (number % 3 == 0) {
        factor = true
        sb.WriteString("Pling")
    }

    if (number % 5 == 0) {
        factor = true
        sb.WriteString("Plang")
    }

    if (number % 7 == 0) {
        factor = true
        sb.WriteString("Plong")
    }

    if (factor == false) {
        return strconv.Itoa(number)
    }

    return sb.String()
}

File: solutions/c/raindrops/1/raindrops.c
#include "raindrops.h"
#include <string.h>
#include <stdio.h>

#define BUFSIZE 16

void convert(char result[], int drops) {

  result[0] = '\0';

  if ((drops % 3) == 0 ){
    strcat(result, "Pling");
  } 

  if ((drops % 5) == 0) {
    strcat(result, "Plang");
  } 

  if ((drops % 7) == 0) {
    strcat(result, "Plong");
  } 

  if ((drops % 3 != 0) && (drops % 5 !=0) && (drops % 7 != 0))
  {
    snprintf(result, BUFSIZE, "%d", drops);
  }

}

